internal/service/resourcepolicy/resource.go

```diff
@@ -0,0 +1,175 @@
+package resourcepolicy
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"regexp"
+
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/mongodb/terraform-provider-mongodbatlas/internal/config"
+	"go.mongodb.org/atlas-sdk/v20240805003/admin"
+)
+
+const resourceName = "resource_policy"
+
+var _ resource.ResourceWithConfigure = &resourcePolicyRS{}
+var _ resource.ResourceWithImportState = &resourcePolicyRS{}
+
+var (
+	errorCreate = "error creating resource"
+	errorRead   = "error reading resource"
+	errorUpdate = "error updating resource"
+)
+
+func Resource() resource.Resource {
+	return &resourcePolicyRS{
+		RSCommon: config.RSCommon{
+			ResourceName: resourceName,
+		},
+	}
+}
+
+type resourcePolicyRS struct {
+	config.RSCommon
+}
+
+func (r *resourcePolicyRS) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = ResourceSchema(ctx)
+}
+
+func (r *resourcePolicyRS) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan TFResourcePolicyModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	orgID := plan.OrgID.ValueString()
+	policies, diags := NewTFPoliciesModelToSDK(ctx, plan.Policies)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	resourcePolicyAPI := connV2.AtlasResourcePoliciesApi
+	policySDK, _, err := resourcePolicyAPI.CreateAtlasResourcePolicy(ctx, orgID, &admin.ApiAtlasResourcePolicyCreate{
+		Name:     plan.Name.ValueStringPointer(),
+		Policies: policies,
+	}).Execute()
+	if err != nil {
+		resp.Diagnostics.AddError(errorCreate, err.Error())
+		return
+	}
+	newResourcePolicyModel, diags := NewTFResourcePolicyModel(ctx, policySDK)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newResourcePolicyModel)...)
+}
+
+func (r *resourcePolicyRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state TFResourcePolicyModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	orgID := state.OrgID.ValueString()
+	resourcePolicyID := state.ID.ValueString()
+	connV2 := r.Client.AtlasV2
+	policySDK, apiResp, err := connV2.AtlasResourcePoliciesApi.GetAtlasResourcePolicy(ctx, orgID, resourcePolicyID).Execute()
+
+	if err != nil {
+		if apiResp != nil && apiResp.StatusCode == http.StatusNotFound {
```

Comment:
apiResp is never nil, no need to check it

Reply:
This is taken straight from the scaffolding. https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/tools/scaffold/template/resource.tmpl#L75
No strong preference...

Reply:
I believe there can be cases where the SDK returns an err without having an apiResp, specifically when failing to create a valid request